extensions/interactions/AlgebraicExpressionInput/directives/oppia-interactive-algebraic-expression-input.component.ts

```diff
@@ -0,0 +1,118 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the AlgebraicExpressionInput interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+require(
+  'interactions/AlgebraicExpressionInput/directives/' +
+  'algebraic-expression-input-rules.service.ts');
+require(
+  'pages/exploration-player-page/services/current-interaction.service.ts');
+
+var nerdamer = require('nerdamer');
+
+angular.module('oppia').component('oppiaInteractiveAlgebraicExpressionInput', {
+  template: require('./algebraic-expression-input-interaction.component.html'),
+  controller: [
+    '$scope', 'CurrentInteractionService',
+    'AlgebraicExpressionInputRulesService',
+    function(
+        $scope, CurrentInteractionService,
+        AlgebraicExpressionInputRulesService) {
+      const ctrl = this;
+      ctrl.value = '';
+      ctrl.hasBeenTouched = false;
+      ctrl.warningText = '';
+
+      ctrl.initializeGuppy = function() {
+        var guppyDivs = document.querySelectorAll('.guppy-div-learner');
+        var divId, guppyInstance;
+        ctrl.hasBeenTouched = false;
+        for (var i = 0; i < guppyDivs.length; i++) {
+          divId = 'guppy_' + Math.floor(Math.random() * 100000000);
+          // Dynamically assigns a unique id to the guppy div.
+          guppyDivs[i].setAttribute('id', divId);
+          // Create a new guppy instance for that div.
+          guppyInstance = new Guppy(divId, {});
+          guppyInstance.event('change', (e) => {
+            ctrl.value = guppyInstance.asciimath();
+            ctrl.hasBeenTouched = true;
+            // Need to manually trigger the digest cycle
+            // to make any 'watchers' aware of changes in ctrl.value.
+            $scope.$apply();
+          });
+        }
+      };
+
+      var cleanErrorMessage = function(errorMessage) {
+        var colonIndex = errorMessage.indexOf(':');
+        if (colonIndex !== -1) {
+          errorMessage = errorMessage.slice(0, colonIndex);
+        }
+        var atColonIndex = errorMessage.indexOf(' at ');
+        if (atColonIndex !== -1) {
+          errorMessage = errorMessage.slice(0, atColonIndex);
+        }
+        if (errorMessage[errorMessage.length - 1] !== '.') {
+          errorMessage += '.';
+        }
+        return errorMessage;
+      };
+
+      ctrl.isCurrentAnswerValid = function() {
+        if (ctrl.hasBeenTouched) {
+          try {
+            var containsVariables = nerdamer(ctrl.value).variables().length > 0;
```

Comment:
This is the only line that needs to be in try - catch to get the error from nerdamer, right? The following errors can be added as warningText directly instead of this block?

Ditto in editor as well.

Reply:
Done.